### ISSUE TITLE
Dataset format

### ISSUE THREAD

**Issue by Zumbalamambo** [dhkim0225/keras-image-segmentation#1]
May in know the format for dataset?

**Comment by dhkim0225** [comment]
Download these files!!

leftImg8bit_trainvaltest.zip (11GB) 
https://www.cityscapes-dataset.com/file-handling/?packageID=3

gtFine_trainvaltest.zip (241MB)
https://www.cityscapes-dataset.com/file-handling/?packageID=1

**Comment by dhkim0225** [comment]
In fact, I did not understand exactly what you were asking. I will write some more information.

We were able to parse the dataset into numpy and create the .h5 file. we could get the data from data.h5  about 10 times faster than opencv imread.

When creating the data, the image was resized(256, 512, 3) and limited to three classes.

**Comment by dhkim0225** [comment]
If you want to train your own dataset,,,,, Sorry but its not supported now. We`ll make methods later(maybe after several months - maybe after we make various sota models)

**Comment by tkwoo** [comment]
I recommend you to use original cityscape dataset structure thuogh we use hdf5 format.

## cityscpae dataset sturcture

The folder structure of the Cityscapes dataset is as follows:
```
{root}/{type}{video}/{split}/{city}/{city}_{seq:0>6}_{frame:0>6}_{type}{ext}
```

The meaning of the individual elements is:
 - `root`  the root folder of the Cityscapes dataset. Many of our scripts check if an environment variable `CITYSCAPES_DATASET` pointing to this folder exists and use this as the default choice.
 - `type`  the type/modality of data, e.g. `gtFine` for fine ground truth, or `leftImg8bit` for left 8-bit images.
 - `split` the split, i.e. train/val/test/train_extra/demoVideo. Note that not all kinds of data exist for all splits. Thus, do not be surprised to occasionally find empty folders.
 - `city`  the city in which this part of the dataset was recorded.
 - `seq`   the sequence number using 6 digits.
 - `frame` the frame number using 6 digits. Note that in some cities very few, albeit very long sequences were recorded, while in some cities many short sequences were recorded, of which only the 19th frame is annotated.
 - `ext`   the extension of the file and optionally a suffix, e.g. `_polygons.json` for ground truth files

Possible values of `type`
 - `gtFine`       the fine annotations, 2975 training, 500 validation, and 1525 testing. This type of annotations is used for validation, testing, and optionally for training. Annotations are encoded using `json` files containing the individual polygons. Additionally, we provide `png` images, where pixel values encode labels. Please refer to `helpers/labels.py` and the scripts in `preparation` for details.
 - `gtCoarse`     the coarse annotations, available for all training and validation images and for another set of 19998 training images (`train_extra`). These annotations can be used for training, either together with gtFine or alone in a weakly supervised setup.
 - `leftImg8bit`  the left images in 8-bit LDR format. These are the standard annotated images.
 - `leftImg16bit` the left images in 16-bit HDR format. These images offer 16 bits per pixel of color depth and contain more information, especially in very dark or bright parts of the scene. Warning: The images are stored as 16-bit pngs, which is non-standard and not supported by all libraries.
 - `rightImg8bit`  the right stereo views in 8-bit LDR format.
 - `rightImg16bit` the right stereo views in 16-bit HDR format.
 - `timestamp`     the time of recording in ns. The first frame of each sequence always has a timestamp of 0.
 - `disparity`     precomputed disparity depth maps. To obtain the disparity values, compute for each pixel p with p > 0: d = ( float(p) - 1. ) / 256., while a value p = 0 is an invalid measurement. Warning: the images are stored as 16-bit pngs, which is non-standard and not supported by all libraries.
 - `camera`        internal and external camera calibration. For details, please refer to [csCalibration.pdf](docs/csCalibration.pdf)
 - `vehicle`       vehicle odometry, GPS coordinates, and outside temperature. For details, please refer to [csCalibration.pdf](docs/csCalibration.pdf)
